common/build_data.h

```diff
@@ -0,0 +1,36 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_COMMON_BUILD_DATA_H_
+#define CARBON_COMMON_BUILD_DATA_H_
+
+#include "common/build_data_linkstamp.h"
+#include "llvm/ADT/StringRef.h"
+
+namespace Carbon::BuildData {
+
+// Build information for a binary, from bazel. Stamped values come from:
+// https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/rules/cpp/CppLinkstampCompileHelper.java
+
+// NOLINTBEGIN(readability-identifier-naming): We want to use constant-style
+// names for the public variables, but cannot use constexpr.
+
+// The platform, per https://bazel.build/extending/platforms.
+static const llvm::StringRef Platform = Internal::platform;
```

Comment:
I think these need to be `inline` instead of `static`. That should let us define and initialize them in the header while letting them be merged between translation units.

Can you use `llvm::StringLiteral` here even without the `constexpr`? Not sure myself, could see it going either way. And a minor point.

Reply:
The `llvm::StringLIteral` I think requires the special compile-time strlen to work, so wouldn't operate here. I'll look at the rest tomorrow. :)

Reply:
Done